### ISSUE TITLE
@BindBean should be able to ignore unknown properties

### ISSUE THREAD

**Issue by qualidafial** [jdbi/jdbi#478]
Migrating a project from Hibernate to JDBI3. When using `@BindBean` on an object with a collection property, `@BindBean` expects that collection to map to something. We need some kind of equivalent to JPA's `@Transient` annotation, or a property on the `@BindBean` annotation itself to let JDBI know it can ignore certain properties.

**Comment by qualidafial** [comment]
Workaround for now: extend `GetHandle` and drop down to the fluent interface in SQL objects. 👎

**Comment by miere** [comment]
Hi @qualidafial. How you doing?

Here at @sizebay we are migrating our main API - which makes heavy use of JDBI2 - to JDBI3 once only the version 3 have _really_ solved the annoying ```@Transaction``` behavior that exists on v2. During the migration process we have faced a lot of problems with "bean"s that have Collection attributes. I do believe it is a feature regression and thus I would consider this issue a bug.

Bellow is the code I did as workaround for this situation (it made sense on my case, once I don't use Collections inside _beans_ as argument for my queries). I hope it help someone else too...
```java
// this class is exposed as SPI for JdbiPlugin
public class ListJDBIArgumentFactory implements ArgumentFactory, JdbiPlugin {

	@Override
	public Optional<Argument> build(Type type, Object o, StatementContext statementContext) {
		return Optional.of( new NullArgument(0) );
	}

	@Override
	public void customizeDbi(Jdbi dbi) {
		dbi.registerArgumentFactory(this);
	}
}
```
As you've suggested, something like ```@Transient``` would be nice at first look. You can apply the same behavior on ```transient```(the modifier) fields . I would also suggest to let 'Java Collection's attribute NULL by default and let developers to change this behavior if needed.

Regards

**Comment by qualidafial** [comment]
We're actually working in another issue on supporting array and collection arguments as SQL arrays, so your approach here would sort of undermine the SQL array support.

I'm considering a few options for resolving this `@BindBean` issue:

1. Just ignore bean properties for which we don't have a mapper. This is similar to what `BeanMapper` does--it only throws an exception if it can't find _any_ columns to map.
2. Introduce an some kind of `@Ignore` annotation and skip any bean property where the getter or setter has that annotation.
3. Add options to the `@BindBean` annotation e.g. `@BindBean(exclude={"things", "widgets"})`.

I dislike option 2 from my own experience with Hibernate--it stinks when the concept of when to include something is baked into the class itself, with no opportunity to override on a case-by-case basis.

I feel like I could call this done with either option 1 or 3, but my gut tells me we should do both of those.

@miere I'd be very interested in any feedback you have on the above.

**Comment by miere** [comment]
From your first set of suggestions I really liked the first one as long as log some debug level information about it. I'm glad we both disagree with the option 2 - entities would become a mess for being crowed of annotations or meta information.

I'm a huge fan of _code conventions_ and _optional configurations_ - for those kind of situations which the convention's default behavior does not fit the developers need. That said, the #494 idea seems great. specially about the fact the variables are bound 'on-demand' (as required by the methods).

**Comment by qualidafial** [comment]
Fixed in #501
